proto/google/events/cloud/build/v1/data.proto

```diff
@@ -0,0 +1,626 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package google.events.cloud.build.v1;
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/timestamp.proto";
+
+option csharp_namespace = "Google.Events.Protobuf.Cloud.Build.V1";
+// Build event data
+// Common build format for Google Cloud Platform API operations.
+// Copied from
+// https://github.com/googleapis/googleapis/blob/master/google/devtools/cloudbuild/v1/cloudbuild.proto,
+message BuildEventData {
+  // Possible status of a build or build step.
+  enum Status {
+    // Status of the build is unknown.
+    STATUS_UNKNOWN = 0;
+
+    // Build or step is queued; work has not yet begun.
+    QUEUED = 1;
+
+    // Build or step is being executed.
+    WORKING = 2;
+
+    // Build or step finished successfully.
+    SUCCESS = 3;
+
+    // Build or step failed to complete successfully.
+    FAILURE = 4;
+
+    // Build or step failed due to an internal cause.
+    INTERNAL_ERROR = 5;
+
+    // Build or step took longer than was allowed.
+    TIMEOUT = 6;
+
+    // Build or step was canceled by a user.
+    CANCELLED = 7;
+
+    // Build was enqueued for longer than the value of `queue_ttl`.
+    EXPIRED = 9;
+  }
+
+  // Unique identifier of the build.
+  string id = 1;
+
+  // ID of the project.
+  string project_id = 16;
+
+  // Status of the build.
+  Status status = 2;
+
+  // Customer-readable message about the current status.
+  string status_detail = 24;
+
+  // The location of the source files to build.
+  Source source = 3;
+
+  // Required. The operations to be performed on the workspace.
+  repeated BuildStep steps = 11;
+
+  // Results of the build.
+  Results results = 10;
+
+  // Time at which the request to create the build was received.
+  google.protobuf.Timestamp create_time = 6;
+
+  // Time at which execution of the build was started.
+  google.protobuf.Timestamp start_time = 7;
+
+  // Time at which execution of the build was finished.
+  //
+  // The difference between finish_time and start_time is the duration of the
+  // build's execution.
+  google.protobuf.Timestamp finish_time = 8;
+
+  // Amount of time that this build should be allowed to run, to second
+  // granularity. If this amount of time elapses, work on the build will cease
+  // and the build status will be `TIMEOUT`.
+  //
+  // Default time is ten minutes.
```

Comment:
Will this always actually be populated, i.e. if the user didn't specify it, it will be present with a value of 10 minutes? If so, let's remove the "default time" part.

Reply:
Yep. If user doesn't specify it, it will be present with a value of 10 minutes.